grakn-engine/src/main/java/ai/grakn/engine/GraknEngineServer.java

```diff
@@ -120,40 +118,53 @@ private TaskManager startTaskManager(String taskManagerClassName) {
     }
 
     public void startHTTP() {
-        configureSpark(spark, port);
+
+        boolean passwordProtected = prop.getPropertyAsBool(GraknEngineConfig.PASSWORD_PROTECTED_PROPERTY, false);
+
+        // TODO: Make sure controllers handle the null case
+        Optional<String> secret = prop.tryProperty(GraknEngineConfig.JWT_SECRET_PROPERTY);
+        @Nullable JWTHandler jwtHandler = secret.map(JWTHandler::create).orElse(null);
+        UsersHandler usersHandler = UsersHandler.create(prop.getProperty(GraknEngineConfig.ADMIN_PASSWORD_PROPERTY), factory);
+
+        configureSpark(spark, prop, jwtHandler);
+
+        // Start the websocket for Graql
+        RemoteSession graqlWebSocket = passwordProtected ? RemoteSession.passwordProtected(usersHandler) : RemoteSession.create();
+        spark.webSocket(REST.WebPath.REMOTE_SHELL_URI, graqlWebSocket);
 
         // Start all the controllers
-        EngineGraknGraphFactory factory = EngineGraknGraphFactory.getInstance();
         new GraqlController(factory, spark);
         new ConceptController(factory, spark);
         new DashboardController(factory, spark);
-        new SystemController(spark);
-        new AuthController(spark);
-        new UserController(spark);
-        new CommitLogController(spark, taskManager);
+        new SystemController(factory, spark);
```

Comment:
This actually makes a lot more sense now. It reduces the cohesion of the controller which will help with making the service orientated architecture we have been talking about. In other words. .  . nice.